opal/mca/btl/tcp/btl_tcp.c

```diff
@@ -531,3 +533,64 @@ void mca_btl_tcp_dump(struct mca_btl_base_module_t* base_btl,
     }
 #endif /* OPAL_ENABLE_DEBUG && WANT_PEER_DUMP */
 }
+
+
+/*
+ * A blocking recv on a non-blocking socket. Used to receive the small
+ * amount of connection information that identifies the endpoints
+ * endpoint.
+ */
+
```

Comment:
I used fcntl to see whether it is a blocking/non-blocking socket. Here is my finding, Send is always non-blocking call.  In case of recv, which is called from two place 
1. mca_btl_tcp_endpoint_recv_connect_ack (from btl_tcp_endpoint.c) - which is non blocking 
2. mca_btl_tcp_component_recv_handler (from btl_tcp_component.c) - which is blocking where we are using SO_RCVTIMEO. 

I will update comment and handle EINPROGRESS (which i missed in recv)